<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Resistant correlations and clustering with the biwt package • biwt</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Resistant correlations and clustering with the biwt package">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">biwt</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/biwt.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/hardin47/biwt/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Resistant correlations and clustering with the biwt package</h1>
                        <h4 data-toc-skip class="author">Frances
Heitkemper, Justine Ouellette, and Johanna Hardin</h4>
            
            <h4 data-toc-skip class="date">September 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/hardin47/biwt/blob/master/vignettes/biwt.Rmd" class="external-link"><code>vignettes/biwt.Rmd</code></a></small>
      <div class="d-none name"><code>biwt.Rmd</code></div>
    </div>

    
    
<!--
vignette: >
  %\VignetteIndexEntry{Resistant correlations and clustering with the biwt package}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
-->
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The vignette walks the reader through calculating resistant
multivariate estimates and correlations. The methods are applied to
presidential election data from 1856 through 2020 to uncover voting
patterns.</p>
<p>The functions in the <strong>biwt</strong> package are based on the
following published work: Hardin, J., Mitani, A., Hicks, L., VanKoten,
B.; <strong>A Robust Measure of Correlation Between Two Genes on a
Microarray</strong>, <em>BMC Bioinformatics</em>, 8:220; 2007.</p>
</div>
<div class="section level2">
<h2 id="biwt-package-functions">
<strong>biwt</strong> package functions<a class="anchor" aria-label="anchor" href="#biwt-package-functions"></a>
</h2>
<div class="section level3">
<h3 id="functions-for-resistant-estimation">Functions for resistant estimation<a class="anchor" aria-label="anchor" href="#functions-for-resistant-estimation"></a>
</h3>
<p>We start by providing the two functions that return the resistant
mean and covariance values from Tukey’s biweight estimation. Note that
the difference between the two functions is the <code><a href="../reference/biwt.est.html">biwt.est()</a></code>
requires a
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>2</mn><mo>×</mo><mi>n</mi></mrow><annotation encoding="application/x-tex">2 \times n</annotation></semantics></math>
matrix or data frame (n is the number of measurements) and
<code><a href="../reference/biwt_est.html">biwt_est()</a></code> requires an
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>n</mi><mo>×</mo><mn>2</mn></mrow><annotation encoding="application/x-tex">n \times 2</annotation></semantics></math>
matrix or data frame (n is the number of observations).</p>
<div class="section level4">
<h4 id="biwt-est">
<code>biwt.est()</code><a class="anchor" aria-label="anchor" href="#biwt-est"></a>
</h4>
<div class="section level5">
<h5 id="biwt-estx-r-0-2-med-init-covmcdx">
<code>biwt.est(x, r = 0.2, med.init = covMcd(x))</code><a class="anchor" aria-label="anchor" href="#biwt-estx-r-0-2-med-init-covmcdx"></a>
</h5>
<p>The function inputs a
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mn>2</mn><mo>×</mo><mi>n</mi></mrow><annotation encoding="application/x-tex">2 \times n</annotation></semantics></math>
matrix, breakdown (percent of data that can be ignored), and initial
estimate of the center and shape of the data. It outputs the measure of
the center and shape of the data using Tukey’s biweight M-estimator.</p>
</div>
</div>
<div class="section level4">
<h4 id="biwt_est">
<code>biwt_est()</code><a class="anchor" aria-label="anchor" href="#biwt_est"></a>
</h4>
<div class="section level5">
<h5 id="biwt_estx-r-med-init">
<code>biwt_est(x, r, med.init)</code><a class="anchor" aria-label="anchor" href="#biwt_estx-r-med-init"></a>
</h5>
<p>The function inputs the same arguments as <code><a href="../reference/biwt.est.html">biwt.est()</a></code>,
but the data input is transposed. That is, the input matrix is
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>n</mi><mo>×</mo><mn>2</mn><mi>.</mi></mrow><annotation encoding="application/x-tex">n \times 2.</annotation></semantics></math>
The output is the same as <code><a href="../reference/biwt.est.html">biwt.est()</a></code>.</p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="functions-for-resistant-correlation">Functions for resistant correlation<a class="anchor" aria-label="anchor" href="#functions-for-resistant-correlation"></a>
</h3>
<p>Based on Tukey’s biweight estimation, <code><a href="../reference/biwt.cor.html">biwt.cor()</a></code> and
<code><a href="../reference/biwt_cor.html">biwt_cor()</a></code> provide resistant estimates of correlation. Note
that the difference between the two functions is that
<code><a href="../reference/biwt.cor.html">biwt.cor()</a></code> requires a
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>g</mi><mo>×</mo><mi>n</mi></mrow><annotation encoding="application/x-tex">g \times n</annotation></semantics></math>
matrix or data frame (n is the number of measurements, g is the number
of observations (genes)) and <code><a href="../reference/biwt_cor.html">biwt_cor()</a></code> requires an
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>n</mi><mo>×</mo><mi>g</mi></mrow><annotation encoding="application/x-tex">n \times g</annotation></semantics></math>
matrix or data frame (n is the number of measurements, g is the number
of observations (genes)).</p>
<div class="section level4">
<h4 id="biwt-cor">
<code>biwt.cor()</code><a class="anchor" aria-label="anchor" href="#biwt-cor"></a>
</h4>
<div class="section level5">
<h5 id="biwt-corx-r-0-2-output-matrix-median-true-full-init-true-absval-true">
<code>biwt.cor(x, r = 0.2, output = "matrix", median = TRUE, full.init = TRUE, absval = TRUE)</code><a class="anchor" aria-label="anchor" href="#biwt-corx-r-0-2-output-matrix-median-true-full-init-true-absval-true"></a>
</h5>
<p>The function inputs a
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>g</mi><mo>×</mo><mi>n</mi></mrow><annotation encoding="application/x-tex">g \times n</annotation></semantics></math>
matrix, the breakdown, intended output, a command to determine whether
the initialization is done using the coordinate-wise median and MAD or
using the minimum covariance determinant (MCD), a argument to determine
whether the initialization is done for each pair separately, and an
argument to determine whether [if the output is a distance] the distance
should be measured as 1 minus the absolute value of the correlation or
as 1 minus the correlation. The output is a matrix, vector, or distance,
depending on the <code>output</code> argument.</p>
</div>
</div>
<div class="section level4">
<h4 id="biwt_cor">
<code>biwt_cor()</code><a class="anchor" aria-label="anchor" href="#biwt_cor"></a>
</h4>
<div class="section level5">
<h5 id="biwt_corx-r-median-true-full-init-true">
<code>biwt_cor(x, r, median = TRUE, full.init = TRUE)</code><a class="anchor" aria-label="anchor" href="#biwt_corx-r-median-true-full-init-true"></a>
</h5>
<p>The function inputs an
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>n</mi><mo>×</mo><mi>g</mi></mrow><annotation encoding="application/x-tex">n \times g</annotation></semantics></math>
matrix, the breakdown, an argument to determine whether the
initialization is done using the coordinate-wise median and MAD or using
the minimum covariance determinant (MCD), and an argument to determine
whether the initialization is done for each pair separately. The
function outputs a vector of
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mo stretchy="true" form="prefix">(</mo><mfrac linethickness="0"><mrow></mrow><mrow><mi>g</mi><mo>,</mo><mn>2</mn></mrow></mfrac><mo stretchy="true" form="postfix">)</mo></mrow><annotation encoding="application/x-tex">\choose{g,2}</annotation></semantics></math>
biweight correlations.</p>
</div>
</div>
<div class="section level4">
<h4 id="biwt_cor_matrix">
<code>biwt_cor_matrix()</code><a class="anchor" aria-label="anchor" href="#biwt_cor_matrix"></a>
</h4>
<div class="section level5">
<h5 id="biwt_cor_matrixx-r-median-true-full-init-true">
<code>biwt_cor_matrix(x, r, median = TRUE, full.init = TRUE)</code><a class="anchor" aria-label="anchor" href="#biwt_cor_matrixx-r-median-true-full-init-true"></a>
</h5>
<p>The function inputs an
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>n</mi><mo>×</mo><mi>g</mi></mrow><annotation encoding="application/x-tex">n \times g</annotation></semantics></math>
matrix, the breakdown, an argument to determine whether the
initialization is done using the coordinate-wise median and MAD or using
the minimum covariance determinant (MCD), and an argument to determine
whether the initialization is done for each pair separately or only one
time at the beginning using the entire data matrix. The function outputs
a
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>g</mi><mo>×</mo><mi>g</mi></mrow><annotation encoding="application/x-tex">g \times g</annotation></semantics></math>
matrix of biweight correlations.</p>
</div>
</div>
<div class="section level4">
<h4 id="biwt_dist_matrix">
<code>biwt_dist_matrix()</code><a class="anchor" aria-label="anchor" href="#biwt_dist_matrix"></a>
</h4>
<div class="section level5">
<h5 id="biwt_dist_matrixx-r-median-true-full-init-true-absval-true">
<code>biwt_dist_matrix(x, r, median = TRUE, full.init = TRUE, absval = TRUE)</code><a class="anchor" aria-label="anchor" href="#biwt_dist_matrixx-r-median-true-full-init-true-absval-true"></a>
</h5>
<p>The function inputs an
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>n</mi><mo>×</mo><mi>g</mi></mrow><annotation encoding="application/x-tex">n \times g</annotation></semantics></math>
matrix, the breakdown, an argument to determine whether the
initialization is done using the coordinate-wise median and MAD or using
the minimum covariance determinant (MCD), an argument to determine
whether the initialization is done for each pair separately, and an
argument to determine whether the distance should be measured as 1 minus
the absolute value of the correlation or as 1 minus the raw correlation.
The function outputs a
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mrow><mi>g</mi><mo>×</mo><mi>g</mi></mrow><annotation encoding="application/x-tex">g \times g</annotation></semantics></math>
matrix of biweight <strong>distances</strong> (either 1 - correlation or
1 - absolute correlation).</p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="correlation">Correlation<a class="anchor" aria-label="anchor" href="#correlation"></a>
</h2>
<p>The goal of the functions in the <strong>biwt</strong> package are to
produce correlations which are resistant to rogue data.</p>
<blockquote>
<p>A <em>resistant</em> method produces results that change only
slightly when a small part of the data is replaced by new numbers,
possibly very different fromt he original ones. [Hoaglin, Mosteller,
Tukey, <strong>Understanding Robust and Exploratory Data
Analysis</strong>, pg 2. 1983]</p>
</blockquote>
<p>For example, the median is a resistant measure of center because even
if some data values are replaced with extreme values, the median will
remain roughly the same since it is the middle number. However, if 50%
of the data are changed, the median’s value will change. The mean is not
a resistant measure of center. If observations are replaced with extreme
values, the mean will be drawn to those values.</p>
<p>Tukey’s biweight gives zero weight to observations that are
sufficiently far from the bulk of the data, thus discounting them in the
calculation of the center, the shape, and the correlation.</p>
<div class="section level3">
<h3 id="biweight-correlation">Biweight Correlation<a class="anchor" aria-label="anchor" href="#biweight-correlation"></a>
</h3>
<p>Biweight correlation is a resistant correlation method used to find
accurate correlation of multivariate data in the presence of outliers.
The biweight correlation functions are set (default) to be resistant for
an outlier proportion of up to 20%. Sufficiently far out outlying points
will be given zero weight, and the correlation is calculated without
those observations, yielding a correlation which is resistant to extreme
values.</p>
</div>
<div class="section level3">
<h3 id="pearson-correlation">Pearson Correlation<a class="anchor" aria-label="anchor" href="#pearson-correlation"></a>
</h3>
<p>Unlike Tukey’s biweight correlation, the Pearson correlation is
calculated with all observations, each getting equal weight. Therefore,
the Pearson correlation is not resistant because extreme value can have
a large impact on the correlation calculation.</p>
</div>
<div class="section level3">
<h3 id="comparison-of-biweight-and-pearson-correlation-on-clean-data">Comparison of Biweight and Pearson correlation on clean data<a class="anchor" aria-label="anchor" href="#comparison-of-biweight-and-pearson-correlation-on-clean-data"></a>
</h3>
<p>When the data are clean, meaning no outliers, both correlation
functions are able to accurately estimate the true population
correlation using a sample. It is also known that Pearson correlation is
the maximum likelihood estimator of the true correlation in
uncontaminated bivariate normal populations (a quality that brings many
nice properties).</p>
<p>We start with clean data (no contamination) and display that for a
single sample, the estimated correlation values are close to the true
parameter of 0.75. Additionally, we show an example of a single sample
generated from a bivariate normal distribution with correlation
parameter of 0.75.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">4747</span><span class="op">)</span></span>
<span><span class="co"># The `contamination` function generates a dataset with `p` percent</span></span>
<span><span class="co"># contaminated data.</span></span>
<span><span class="co"># `n`: number of observations</span></span>
<span><span class="co"># `p`: percent of contaminated observations</span></span>
<span><span class="co"># `pc`: correlation of the population from which the data are generated</span></span>
<span><span class="co"># `u1x`: lower bound of uniform dist on x-axis (contaminated data)</span></span>
<span><span class="co"># `u2x`: upper bound of uniform dist on x-axis (contaminated data)</span></span>
<span><span class="co"># `u1y`: lower bound of uniform dist on y-axis (contaminated data)</span></span>
<span><span class="co"># `u2y`: upper bound of uniform dist on y-axis (contaminated data)</span></span>
<span><span class="va">contamination</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">n</span> <span class="op">=</span> <span class="fl">100</span>, <span class="va">p</span> <span class="op">=</span> <span class="fl">0.1</span>, <span class="va">pc</span> <span class="op">=</span> <span class="fl">0.75</span>, </span>
<span>                          <span class="va">u1x</span> <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, <span class="va">u2x</span> <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, <span class="va">u1y</span> <span class="op">=</span> <span class="fl">15</span>, <span class="va">u2y</span> <span class="op">=</span> <span class="fl">25</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="va">temp2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu">MASS</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/MASS/man/mvrnorm.html" class="external-link">mvrnorm</a></span><span class="op">(</span><span class="va">n</span><span class="op">*</span><span class="op">(</span><span class="fl">1</span><span class="op">-</span><span class="va">p</span><span class="op">)</span>, mu <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>, </span>
<span>                          Sigma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="va">pc</span>,<span class="va">pc</span>,<span class="fl">1</span><span class="op">)</span>, ncol<span class="op">=</span> <span class="fl">2</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, </span>
<span>               <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>X1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">n</span><span class="op">*</span><span class="va">p</span>, <span class="va">u1x</span>, <span class="va">u2x</span><span class="op">)</span>, X2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="va">n</span><span class="op">*</span><span class="va">p</span>, <span class="va">u1y</span>, <span class="va">u2y</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">temp2</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span></span>
<span><span class="co"># The `correlation_func` function simulates contaminated data</span></span>
<span><span class="co"># and returns both the biweight and the Pearson correlation</span></span>
<span><span class="co"># `r`: the breakdown (% of allowed contamination)</span></span>
<span><span class="va">correlation_func</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">r</span> <span class="op">=</span> <span class="fl">0.2</span>, <span class="va">n</span> <span class="op">=</span> <span class="fl">100</span>, <span class="va">p</span> <span class="op">=</span> <span class="fl">0.1</span>, <span class="va">pc</span> <span class="op">=</span> <span class="fl">0.75</span>, </span>
<span>                             <span class="va">u1x</span> <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, <span class="va">u2x</span> <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, <span class="va">u1y</span> <span class="op">=</span> <span class="fl">15</span>, <span class="va">u2y</span> <span class="op">=</span> <span class="fl">25</span><span class="op">)</span><span class="op">{</span></span>
<span>  </span>
<span>  <span class="va">samp_data</span> <span class="op">&lt;-</span> <span class="fu">contamination</span><span class="op">(</span>n <span class="op">=</span> <span class="va">n</span>, p <span class="op">=</span> <span class="va">p</span>, pc <span class="op">=</span> <span class="va">pc</span>, </span>
<span>                             u1x <span class="op">=</span> <span class="va">u1x</span>, u2x <span class="op">=</span> <span class="va">u2x</span>, u1y <span class="op">=</span> <span class="va">u1y</span>, u2y <span class="op">=</span> <span class="va">u2y</span><span class="op">)</span></span>
<span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>bwcorrelation <span class="op">=</span> <span class="fu"><a href="../reference/biwt_cor.html">biwt_cor</a></span><span class="op">(</span><span class="va">samp_data</span>, <span class="va">r</span><span class="op">)</span><span class="op">$</span><span class="va">biwt_corr</span>, </span>
<span>                    pcorrelation <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="va">samp_data</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="co"># First, run the correlation function on clean (uncontaminated) data</span></span>
<span><span class="fu">correlation_func</span><span class="op">(</span>p <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">gt</span><span class="fu">::</span><span class="fu"><a href="https://gt.rstudio.com/reference/gt.html" class="external-link">gt</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Biweight and Pearson correlations from a sample of 100 observations in an uncontaminated bivariate normal distribution with population correlation of 0.75."</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">gt</span><span class="fu">::</span><span class="fu"><a href="https://gt.rstudio.com/reference/cols_label.html" class="external-link">cols_label</a></span><span class="op">(</span>bwcorrelation <span class="op">=</span> <span class="st">"biweight correlation"</span>, </span>
<span>             pcorrelation <span class="op">=</span> <span class="st">"Pearson correlation"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">gt</span><span class="fu">::</span><span class="fu"><a href="https://gt.rstudio.com/reference/tab_options.html" class="external-link">tab_options</a></span><span class="op">(</span>table.width <span class="op">=</span> <span class="fu">gt</span><span class="fu">::</span><span class="fu"><a href="https://gt.rstudio.com/reference/pct.html" class="external-link">pct</a></span><span class="op">(</span><span class="fl">85</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div id="lhujjymwyb" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#lhujjymwyb table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#lhujjymwyb thead, #lhujjymwyb tbody, #lhujjymwyb tfoot, #lhujjymwyb tr, #lhujjymwyb td, #lhujjymwyb th {
  border-style: none;
}

#lhujjymwyb p {
  margin: 0;
  padding: 0;
}

#lhujjymwyb .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: 85%;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#lhujjymwyb .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#lhujjymwyb .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#lhujjymwyb .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#lhujjymwyb .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#lhujjymwyb .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#lhujjymwyb .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#lhujjymwyb .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#lhujjymwyb .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#lhujjymwyb .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#lhujjymwyb .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#lhujjymwyb .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#lhujjymwyb .gt_spanner_row {
  border-bottom-style: hidden;
}

#lhujjymwyb .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#lhujjymwyb .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#lhujjymwyb .gt_from_md > :first-child {
  margin-top: 0;
}

#lhujjymwyb .gt_from_md > :last-child {
  margin-bottom: 0;
}

#lhujjymwyb .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#lhujjymwyb .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#lhujjymwyb .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#lhujjymwyb .gt_row_group_first td {
  border-top-width: 2px;
}

#lhujjymwyb .gt_row_group_first th {
  border-top-width: 2px;
}

#lhujjymwyb .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#lhujjymwyb .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#lhujjymwyb .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#lhujjymwyb .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#lhujjymwyb .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#lhujjymwyb .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#lhujjymwyb .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#lhujjymwyb .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#lhujjymwyb .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#lhujjymwyb .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#lhujjymwyb .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#lhujjymwyb .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#lhujjymwyb .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#lhujjymwyb .gt_left {
  text-align: left;
}

#lhujjymwyb .gt_center {
  text-align: center;
}

#lhujjymwyb .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#lhujjymwyb .gt_font_normal {
  font-weight: normal;
}

#lhujjymwyb .gt_font_bold {
  font-weight: bold;
}

#lhujjymwyb .gt_font_italic {
  font-style: italic;
}

#lhujjymwyb .gt_super {
  font-size: 65%;
}

#lhujjymwyb .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#lhujjymwyb .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#lhujjymwyb .gt_indent_1 {
  text-indent: 5px;
}

#lhujjymwyb .gt_indent_2 {
  text-indent: 10px;
}

#lhujjymwyb .gt_indent_3 {
  text-indent: 15px;
}

#lhujjymwyb .gt_indent_4 {
  text-indent: 20px;
}

#lhujjymwyb .gt_indent_5 {
  text-indent: 25px;
}

#lhujjymwyb .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#lhujjymwyb div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="table gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<caption>Biweight and Pearson correlations from a sample of 100 observations in an uncontaminated bivariate normal distribution with population correlation of 0.75.</caption>
  <thead><tr class="gt_col_headings">
<th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="biweight correlation">biweight correlation</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Pearson correlation">Pearson correlation</th>
    </tr></thead>
<tbody class="gt_table_body"><tr>
<td headers="bwcorrelation" class="gt_row gt_right">0.7316408</td>
<td headers="pcorrelation" class="gt_row gt_right">0.728225</td>
</tr></tbody>
</table>
</div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">contamination</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">100</span>, p <span class="op">=</span> <span class="fl">0</span>, pc <span class="op">=</span> <span class="fl">0.75</span>, u1x <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, u2x <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, u1y <span class="op">=</span> <span class="fl">15</span>, u2y <span class="op">=</span> <span class="fl">25</span><span class="op">)</span> <span class="op">|&gt;</span>   </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">X1</span>, y <span class="op">=</span> <span class="va">X2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"first coordinant"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"second coordinant"</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"Uncontaminated dataset"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-3-1.png" alt="Plot of uncontaminated bivariate normal sample of size 100 from a population with true correlation of 0.75." width="729.166666666667"><p class="caption">
Plot of uncontaminated bivariate normal sample of size 100 from a
population with true correlation of 0.75.
</p>
</div>
</div>
<div class="section level3">
<h3 id="comparison-of-biweight-and-pearson-correlation-on-contaminated-data">Comparison of Biweight and Pearson correlation on contaminated
data<a class="anchor" aria-label="anchor" href="#comparison-of-biweight-and-pearson-correlation-on-contaminated-data"></a>
</h3>
<p>When the data are contaminated, the biweight correlation is able to
estimate the underlying population correlation more accurately than the
Pearson correlation does. The biweight down weights the contaminated
data and calculates the correlation on the remaining uncontaminated
values.</p>
<p>In the example below, 10% of the data are replaced with contamination
which is distributed uniformly in the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>x</mi><annotation encoding="application/x-tex">x</annotation></semantics></math>
direction from -3 to -1 and uniformly in the
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>y</mi><annotation encoding="application/x-tex">y</annotation></semantics></math>
direction from 15 to 25. We provide the correlation values (both
biweight and Pearson) on the contaminated data and note that the
biweight value is much closer to the population parameter (0.75) than
the Pearson correlation is. Additionally, we show an example of a single
sample generated from a contaminated (10% of observations) bivariate
normal distribution with correlation parameter of 0.75.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># using the default values</span></span>
<span><span class="fu">correlation_func</span><span class="op">(</span>r <span class="op">=</span> <span class="fl">0.2</span>, n <span class="op">=</span> <span class="fl">100</span>, p <span class="op">=</span> <span class="fl">0.1</span>, pc <span class="op">=</span> <span class="fl">0.75</span>,</span>
<span>                 u1x <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, u2x <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, u1y <span class="op">=</span> <span class="fl">15</span>, u2y <span class="op">=</span> <span class="fl">25</span><span class="op">)</span><span class="op">|&gt;</span> </span>
<span>  <span class="fu">gt</span><span class="fu">::</span><span class="fu"><a href="https://gt.rstudio.com/reference/gt.html" class="external-link">gt</a></span><span class="op">(</span>caption <span class="op">=</span> <span class="st">"Biweight and Pearson correlations from a sample of 100 observations in a 10% contaminated bivariate normal distribution with population correlation of 0.75."</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">gt</span><span class="fu">::</span><span class="fu"><a href="https://gt.rstudio.com/reference/cols_label.html" class="external-link">cols_label</a></span><span class="op">(</span>bwcorrelation <span class="op">=</span> <span class="st">"biweight correlation"</span>, </span>
<span>             pcorrelation <span class="op">=</span> <span class="st">"Pearson correlation"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">gt</span><span class="fu">::</span><span class="fu"><a href="https://gt.rstudio.com/reference/tab_options.html" class="external-link">tab_options</a></span><span class="op">(</span>table.width <span class="op">=</span> <span class="fu">gt</span><span class="fu">::</span><span class="fu"><a href="https://gt.rstudio.com/reference/pct.html" class="external-link">pct</a></span><span class="op">(</span><span class="fl">85</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div id="qfygufyadf" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#qfygufyadf table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#qfygufyadf thead, #qfygufyadf tbody, #qfygufyadf tfoot, #qfygufyadf tr, #qfygufyadf td, #qfygufyadf th {
  border-style: none;
}

#qfygufyadf p {
  margin: 0;
  padding: 0;
}

#qfygufyadf .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: 85%;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#qfygufyadf .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#qfygufyadf .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#qfygufyadf .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#qfygufyadf .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#qfygufyadf .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qfygufyadf .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#qfygufyadf .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#qfygufyadf .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#qfygufyadf .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#qfygufyadf .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#qfygufyadf .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#qfygufyadf .gt_spanner_row {
  border-bottom-style: hidden;
}

#qfygufyadf .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#qfygufyadf .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#qfygufyadf .gt_from_md > :first-child {
  margin-top: 0;
}

#qfygufyadf .gt_from_md > :last-child {
  margin-bottom: 0;
}

#qfygufyadf .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#qfygufyadf .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#qfygufyadf .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#qfygufyadf .gt_row_group_first td {
  border-top-width: 2px;
}

#qfygufyadf .gt_row_group_first th {
  border-top-width: 2px;
}

#qfygufyadf .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#qfygufyadf .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#qfygufyadf .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#qfygufyadf .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qfygufyadf .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#qfygufyadf .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#qfygufyadf .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#qfygufyadf .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#qfygufyadf .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qfygufyadf .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#qfygufyadf .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#qfygufyadf .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#qfygufyadf .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#qfygufyadf .gt_left {
  text-align: left;
}

#qfygufyadf .gt_center {
  text-align: center;
}

#qfygufyadf .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#qfygufyadf .gt_font_normal {
  font-weight: normal;
}

#qfygufyadf .gt_font_bold {
  font-weight: bold;
}

#qfygufyadf .gt_font_italic {
  font-style: italic;
}

#qfygufyadf .gt_super {
  font-size: 65%;
}

#qfygufyadf .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#qfygufyadf .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#qfygufyadf .gt_indent_1 {
  text-indent: 5px;
}

#qfygufyadf .gt_indent_2 {
  text-indent: 10px;
}

#qfygufyadf .gt_indent_3 {
  text-indent: 15px;
}

#qfygufyadf .gt_indent_4 {
  text-indent: 20px;
}

#qfygufyadf .gt_indent_5 {
  text-indent: 25px;
}

#qfygufyadf .katex-display {
  display: inline-flex !important;
  margin-bottom: 0.75em !important;
}

#qfygufyadf div.Reactable > div.rt-table > div.rt-thead > div.rt-tr.rt-tr-group-header > div.rt-th-group:after {
  height: 0px !important;
}
</style>
<table class="table gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<caption>Biweight and Pearson correlations from a sample of 100 observations in a 10% contaminated bivariate normal distribution with population correlation of 0.75.</caption>
  <thead><tr class="gt_col_headings">
<th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="biweight correlation">biweight correlation</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1" scope="col" id="Pearson correlation">Pearson correlation</th>
    </tr></thead>
<tbody class="gt_table_body"><tr>
<td headers="bwcorrelation" class="gt_row gt_right">0.7209249</td>
<td headers="pcorrelation" class="gt_row gt_right">-0.4179869</td>
</tr></tbody>
</table>
</div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># using the default values</span></span>
<span><span class="fu">contamination</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">100</span>, p <span class="op">=</span> <span class="fl">0.1</span>, pc <span class="op">=</span> <span class="fl">0.75</span>, </span>
<span>              u1x <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, u2x <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, u1y <span class="op">=</span> <span class="fl">15</span>, u2y <span class="op">=</span> <span class="fl">25</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">X1</span>, y <span class="op">=</span> <span class="va">X2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"first coordinant"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"second coordinant"</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"Dataset with 10% contamination"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-5-1.png" alt="Plot of 10% contaminated bivariate normal sample of size 100 from a population with true correlation of 0.75." width="729.166666666667"><p class="caption">
Plot of 10% contaminated bivariate normal sample of size 100 from a
population with true correlation of 0.75.
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="simulation-study">Simulation Study<a class="anchor" aria-label="anchor" href="#simulation-study"></a>
</h2>
<p>In the following simulation study, we investigate different
contamination levels and assess the resulting biweight and Pearson
correlations.</p>
<div class="section level4">
<h4 id="example-1-breakdown-contamination-20-compressed-contamination">Example #1: breakdown &gt; contamination, 20% compressed
contamination<a class="anchor" aria-label="anchor" href="#example-1-breakdown-contamination-20-compressed-contamination"></a>
</h4>
<p>When the breakdown (r = 0.3) is greater than the percentage of
contamination (p = 0.2), then the contaminated data is down weighted
during the biweight correlation computation. The result is that the
biweight correlation estimates are more accurate than the Pearson
correlation estimates.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data1</span> <span class="op">&lt;-</span> <span class="fu">contamination</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">100</span>, p <span class="op">=</span> <span class="fl">0.2</span>, pc <span class="op">=</span> <span class="fl">0.75</span>, </span>
<span>                      u1x <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, u2x <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, u1y <span class="op">=</span> <span class="fl">15</span>, u2y <span class="op">=</span> <span class="fl">25</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data1</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">X1</span>, y <span class="op">=</span> <span class="va">X2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"first coordinant"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"second coordinant"</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"Dataset with 20% compressed contamination"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-6-1.png" alt="One simulated dataset with n = 100, 20% compressed contamination, and a population correlation of 0.75." width="729.166666666667"><p class="caption">
One simulated dataset with n = 100, 20% compressed contamination, and a
population correlation of 0.75.
</p>
</div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">correlation1</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">500</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">map_dfr</span><span class="op">(</span><span class="op">~</span><span class="fu">correlation_func</span><span class="op">(</span>r <span class="op">=</span> <span class="fl">0.3</span>, n <span class="op">=</span> <span class="fl">100</span>, p <span class="op">=</span> <span class="fl">0.2</span>, pc <span class="op">=</span> <span class="fl">0.75</span>, </span>
<span>                            u1x <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, u2x <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, u1y <span class="op">=</span> <span class="fl">15</span>, u2y <span class="op">=</span> <span class="fl">25</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">correlation_long1</span> <span class="op">&lt;-</span> <span class="va">correlation1</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span> <span class="op">(</span><span class="st">"bwcorrelation"</span>, <span class="st">"pcorrelation"</span><span class="op">)</span>, </span>
<span>               names_to <span class="op">=</span> <span class="st">"statistic"</span>, values_to <span class="op">=</span> <span class="st">"value"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">correlation_long1</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>statistic <span class="op">=</span> <span class="fu">case_when</span><span class="op">(</span></span>
<span>    <span class="va">statistic</span> <span class="op">==</span> <span class="st">"bwcorrelation"</span> <span class="op">~</span> <span class="st">"biweight correlation"</span>,</span>
<span>    <span class="va">statistic</span> <span class="op">==</span> <span class="st">"pcorrelation"</span> <span class="op">~</span> <span class="st">"Pearson correlation"</span></span>
<span>  <span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0.75</span>, color <span class="op">=</span> <span class="st">"pink"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">statistic</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"correlation value"</span>, y <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-7-1.png" alt="Histogram of correlations for 500 random samples.  Each of the 500 datasets has n = 100 and 20% compressed contamination. The pink vertical line is at the population correlation, 0.75." width="729.166666666667"><p class="caption">
Histogram of correlations for 500 random samples. Each of the 500
datasets has n = 100 and 20% compressed contamination. The pink vertical
line is at the population correlation, 0.75.
</p>
</div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">correlation1</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">bwcorrelation</span>, y <span class="op">=</span> <span class="va">pcorrelation</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0.75</span>, color <span class="op">=</span> <span class="st">"pink"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0.75</span>, color <span class="op">=</span> <span class="st">"pink"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"biweight correlation"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"Pearson correlation"</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"biweight and Pearson correlation for each \nof 500 simulated datasets"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-8-1.png" alt="Scatterplot describing the biweight and Pearson correlations calculated for each of 500 contaminated datasets. The pink vertical and horizontal lines are at the population correlation, 0.75." width="729.166666666667"><p class="caption">
Scatterplot describing the biweight and Pearson correlations calculated
for each of 500 contaminated datasets. The pink vertical and horizontal
lines are at the population correlation, 0.75.
</p>
</div>
</div>
<div class="section level4">
<h4 id="example-2-breakdown-contamination-20-diffuse-contamination">Example #2: breakdown &gt; contamination, 20% diffuse
contamination<a class="anchor" aria-label="anchor" href="#example-2-breakdown-contamination-20-diffuse-contamination"></a>
</h4>
<p>By changing the range from which the outlying contamination comes
from, we can see the impact that contamination has on both the biweight
and the Pearson correlations. Despite the contamination being quite
diffuse, the biweight is able to estimate the correlation to be close to
the true value of 0.75. Note that we would expect the biweight
correlation to be accurate because the percentage of contamination is
less than the breakdown. The Pearson correlation does better on the
diffuse correlation than on the compressed correlation, but the Pearson
estimates are still not close to the true value of 0.75. The compressed
contamination provided Pearson correlation estimates close to -0.5,
while the diffuse contamination Pearson correlation estimates are
centered roughly around zero.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data2</span> <span class="op">&lt;-</span> <span class="fu">contamination</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">100</span>, p <span class="op">=</span> <span class="fl">0.2</span>, pc <span class="op">=</span> <span class="fl">0.75</span>, </span>
<span>                       u1x <span class="op">=</span> <span class="op">-</span><span class="fl">10</span>, u2x <span class="op">=</span> <span class="fl">10</span>, u1y <span class="op">=</span> <span class="op">-</span><span class="fl">10</span>, u2y <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data2</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">X1</span>, y <span class="op">=</span> <span class="va">X2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"first coordinant"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"second coordinant"</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"Dataset with 20% diffuse contamination"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-9-1.png" alt="One simulated dataset with n = 100, 20% diffuse contamination, and a population correlation of 0.75." width="729.166666666667"><p class="caption">
One simulated dataset with n = 100, 20% diffuse contamination, and a
population correlation of 0.75.
</p>
</div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">correlation2</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">500</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">map_dfr</span><span class="op">(</span><span class="op">~</span><span class="fu">correlation_func</span><span class="op">(</span>r <span class="op">=</span> <span class="fl">0.3</span>, n <span class="op">=</span> <span class="fl">100</span>, p <span class="op">=</span> <span class="fl">0.2</span>, pc <span class="op">=</span> <span class="fl">0.75</span>, </span>
<span>                            u1x <span class="op">=</span> <span class="op">-</span><span class="fl">10</span>, u2x <span class="op">=</span> <span class="fl">10</span>, u1y <span class="op">=</span> <span class="op">-</span><span class="fl">10</span>, u2y <span class="op">=</span> <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">correlation_long2</span> <span class="op">&lt;-</span> <span class="va">correlation2</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"bwcorrelation"</span>, <span class="st">"pcorrelation"</span><span class="op">)</span>, </span>
<span>               names_to <span class="op">=</span> <span class="st">"statistic"</span>, values_to <span class="op">=</span> <span class="st">"val"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">correlation_long2</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>statistic <span class="op">=</span> <span class="fu">case_when</span><span class="op">(</span></span>
<span>    <span class="va">statistic</span> <span class="op">==</span> <span class="st">"bwcorrelation"</span> <span class="op">~</span> <span class="st">"biweight correlation"</span>,</span>
<span>    <span class="va">statistic</span> <span class="op">==</span> <span class="st">"pcorrelation"</span> <span class="op">~</span> <span class="st">"Pearson correlation"</span></span>
<span>  <span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">val</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0.75</span>, color <span class="op">=</span> <span class="st">"pink"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">statistic</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"correlation value"</span>, y <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-10-1.png" alt="Histogram of correlations for 500 random samples.  Each of the 500 datasets has n = 100 and 20% diffuse contamination. The pink vertical line is at the population correlation, 0.75." width="729.166666666667"><p class="caption">
Histogram of correlations for 500 random samples. Each of the 500
datasets has n = 100 and 20% diffuse contamination. The pink vertical
line is at the population correlation, 0.75.
</p>
</div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">correlation2</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">bwcorrelation</span>, y <span class="op">=</span> <span class="va">pcorrelation</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0.75</span>, color <span class="op">=</span> <span class="st">"pink"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0.75</span>, color <span class="op">=</span> <span class="st">"pink"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"biweight correlation"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"Pearson correlation"</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"biweight and Pearson correlation for each \nof 500 simulated datasets"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-11-1.png" alt="Scatterplot describing the biweight and Pearson correlations calculated for each of 500 20% diffuse contaminated datasets. The pink vertical and horizontal lines are at the population correlation, 0.75." width="729.166666666667"><p class="caption">
Scatterplot describing the biweight and Pearson correlations calculated
for each of 500 20% diffuse contaminated datasets. The pink vertical and
horizontal lines are at the population correlation, 0.75.
</p>
</div>
</div>
<div class="section level4">
<h4 id="example-3-breakdown-contamination-40-compressed-contamination">Example #3: breakdown &lt; contamination, 40% compressed
contamination<a class="anchor" aria-label="anchor" href="#example-3-breakdown-contamination-40-compressed-contamination"></a>
</h4>
<p>The next example retains the parameters from Example #1 with one
change. The proportion of contamination is increased to 40%. Recall that
the breakdown is set at 30%, so the contamination is higher than the
breakdown. Given that the contamination is higher than the breakdown,
the biweight is no longer able to estimate the correlation close to the
true parameter of 0.75. The biweight is set to consider (i.e.,
downweight) no more than 30% of the observations as outlying, and so 40%
contamination sets the some of the contamination to be part of the bulk
of the observations. As with other contamination values, the Pearson
correlation is unable to accurately estimate the true parameter of
0.75.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data3</span> <span class="op">&lt;-</span> <span class="fu">contamination</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">100</span>, p <span class="op">=</span> <span class="fl">.4</span>, pc <span class="op">=</span> <span class="fl">0.75</span>, </span>
<span>                       u1x <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, u2x <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, u1y <span class="op">=</span> <span class="fl">15</span>, u2y <span class="op">=</span> <span class="fl">25</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data3</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">X1</span>, y <span class="op">=</span> <span class="va">X2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"first coordinant"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"second coordinant"</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"Dataset with 40% compressed contamination"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-12-1.png" alt="One simulated dataset with n = 100, 40% compressed contamination, and a population correlation of 0.75." width="729.166666666667"><p class="caption">
One simulated dataset with n = 100, 40% compressed contamination, and a
population correlation of 0.75.
</p>
</div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">correlation3</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">500</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">map_dfr</span><span class="op">(</span><span class="op">~</span><span class="fu">correlation_func</span><span class="op">(</span>r <span class="op">=</span> <span class="fl">0.3</span>, n <span class="op">=</span> <span class="fl">100</span>, p <span class="op">=</span> <span class="fl">.4</span>, pc <span class="op">=</span> <span class="fl">0.75</span>, </span>
<span>                            u1x <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, u2x <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, u1y <span class="op">=</span> <span class="fl">15</span>, u2y <span class="op">=</span> <span class="fl">25</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">correlation_long3</span> <span class="op">&lt;-</span> <span class="va">correlation3</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"bwcorrelation"</span>, <span class="st">"pcorrelation"</span><span class="op">)</span>, </span>
<span>               names_to <span class="op">=</span> <span class="st">"statistic"</span>, values_to <span class="op">=</span> <span class="st">"value"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">correlation_long3</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>statistic <span class="op">=</span> <span class="fu">case_when</span><span class="op">(</span></span>
<span>    <span class="va">statistic</span> <span class="op">==</span> <span class="st">"bwcorrelation"</span> <span class="op">~</span> <span class="st">"biweight correlation"</span>,</span>
<span>    <span class="va">statistic</span> <span class="op">==</span> <span class="st">"pcorrelation"</span> <span class="op">~</span> <span class="st">"Pearson correlation"</span></span>
<span>  <span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0.75</span>, color <span class="op">=</span> <span class="st">"pink"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">statistic</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"correlation value"</span>, y <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-13-1.png" alt="Histogram of correlations for 500 random samples.  Each of the 500 datasets has n = 100 and 40% compressed contamination. The pink vertical line is at the population correlation, 0.75." width="729.166666666667"><p class="caption">
Histogram of correlations for 500 random samples. Each of the 500
datasets has n = 100 and 40% compressed contamination. The pink vertical
line is at the population correlation, 0.75.
</p>
</div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">correlation3</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">bwcorrelation</span>, y <span class="op">=</span> <span class="va">pcorrelation</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0.75</span>, color <span class="op">=</span> <span class="st">"pink"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0.75</span>, color <span class="op">=</span> <span class="st">"pink"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"biweight correlation"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"Pearson correlation"</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"biweight and Pearson correlation for each \nof 500 simulated datasets"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-14-1.png" alt="Scatterplot describing the biweight and Pearson correlations calculated for each of 500 40% compressed contaminated datasets. The pink vertical and horizontal lines are at the population correlation, 0.75." width="729.166666666667"><p class="caption">
Scatterplot describing the biweight and Pearson correlations calculated
for each of 500 40% compressed contaminated datasets. The pink vertical
and horizontal lines are at the population correlation, 0.75.
</p>
</div>
</div>
<div class="section level4">
<h4 id="example-4-low-population-correlation-parameter-of-0-1-20-compressed-contamination">Example #4: low population correlation parameter of 0.1, 20%
compressed contamination<a class="anchor" aria-label="anchor" href="#example-4-low-population-correlation-parameter-of-0-1-20-compressed-contamination"></a>
</h4>
<p>One last change allows to further compare the biweight and Pearson
correlations. Again, all parameter settings are the same from Example #1
with one change. The true population correlation is set to 0.1 (a much
weaker correlation). We see that the value of the population correlation
does not have a strong effect on the ability of the biweight to estimate
the correlation well and the Pearson to estmate the correlation poorly.
The results in Example #4 are quite similar to the results in Example
#1.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data4</span> <span class="op">&lt;-</span> <span class="fu">contamination</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">100</span>, p <span class="op">=</span> <span class="fl">0.2</span>, pc <span class="op">=</span> <span class="fl">0.1</span>, </span>
<span>                       u1x <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, u2x <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, u1y <span class="op">=</span> <span class="fl">15</span>, u2y <span class="op">=</span> <span class="fl">25</span><span class="op">)</span></span>
<span></span>
<span><span class="va">data4</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">X1</span>, y <span class="op">=</span> <span class="va">X2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"first coordinant"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"second coordinant"</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"Dataset with 20% compressed contamination"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-15-1.png" alt="One simulated dataset with n = 100, 20% compressed contamination, and a population correlation of 0.1." width="729.166666666667"><p class="caption">
One simulated dataset with n = 100, 20% compressed contamination, and a
population correlation of 0.1.
</p>
</div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">correlation4</span> <span class="op">&lt;-</span> <span class="fl">1</span><span class="op">:</span><span class="fl">500</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">map_dfr</span><span class="op">(</span><span class="op">~</span><span class="fu">correlation_func</span><span class="op">(</span>r <span class="op">=</span> <span class="fl">0.3</span>, n <span class="op">=</span> <span class="fl">100</span>, p <span class="op">=</span> <span class="fl">0.2</span>, pc <span class="op">=</span> <span class="fl">0.1</span>, </span>
<span>                            u1x <span class="op">=</span> <span class="op">-</span><span class="fl">3</span>, u2x <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, u1y <span class="op">=</span> <span class="fl">15</span>, u2y <span class="op">=</span> <span class="fl">25</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">correlation_long4</span> <span class="op">&lt;-</span> <span class="va">correlation4</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">pivot_longer</span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span> <span class="op">(</span><span class="st">"bwcorrelation"</span>, <span class="st">"pcorrelation"</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"statistic"</span>, values_to <span class="op">=</span> <span class="st">"value"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">correlation_long4</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span>statistic <span class="op">=</span> <span class="fu">case_when</span><span class="op">(</span></span>
<span>    <span class="va">statistic</span> <span class="op">==</span> <span class="st">"bwcorrelation"</span> <span class="op">~</span> <span class="st">"biweight correlation"</span>,</span>
<span>    <span class="va">statistic</span> <span class="op">==</span> <span class="st">"pcorrelation"</span> <span class="op">~</span> <span class="st">"Pearson correlation"</span></span>
<span>  <span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">value</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_histogram</span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0.1</span>, color <span class="op">=</span> <span class="st">"pink"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">facet_wrap</span><span class="op">(</span><span class="op">~</span><span class="va">statistic</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"correlation value"</span>, y <span class="op">=</span> <span class="st">""</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-16-1.png" alt="Histogram of correlations for 500 random samples.  Each of the 500 datasets has n = 100 and 20% compressed contamination. The pink vertical line is at the population correlation, 0.1." width="729.166666666667"><p class="caption">
Histogram of correlations for 500 random samples. Each of the 500
datasets has n = 100 and 20% compressed contamination. The pink vertical
line is at the population correlation, 0.1.
</p>
</div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">correlation4</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">ggplot</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_point</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>x <span class="op">=</span> <span class="va">bwcorrelation</span>, y <span class="op">=</span> <span class="va">pcorrelation</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_vline</span><span class="op">(</span>xintercept <span class="op">=</span> <span class="fl">0.1</span>, color <span class="op">=</span> <span class="st">"pink"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0.1</span>, color <span class="op">=</span> <span class="st">"pink"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu">theme_minimal</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">labs</span><span class="op">(</span>x <span class="op">=</span> <span class="st">"biweight correlation"</span>,</span>
<span>       y <span class="op">=</span> <span class="st">"Pearson correlation"</span>,</span>
<span>       title <span class="op">=</span> <span class="st">"biweight and Pearson correlation for each \nof 500 simulated datasets"</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="biwt_files/figure-html/unnamed-chunk-17-1.png" alt="Scatterplot describing the biweight and Pearson correlations calculated for each of 500 contaminated datasets. The pink vertical and horizontal lines are at the population correlation, 0.1." width="729.166666666667"><p class="caption">
Scatterplot describing the biweight and Pearson correlations calculated
for each of 500 contaminated datasets. The pink vertical and horizontal
lines are at the population correlation, 0.1.
</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="presidential-candidate-voting-data">Presidential Candidate Voting Data<a class="anchor" aria-label="anchor" href="#presidential-candidate-voting-data"></a>
</h2>
<p>The following clustering example models data on every US presidential
election between 1856 and 2020. Each row of the dataset is one of the 50
united states. Each column represents the percentage of votes for the
Republican candidate in the election, over each of the 42 elections from
1856 to 2020. Note that we are looking to cluster the states, i.e., the
rows (using information from the years = columns). To that end, the main
<strong>biwt</strong> R package functions to be used is
<code><a href="../reference/biwt.cor.html">biwt.cor()</a></code>.</p>
<p>The data from 1856-1972 are provided in the <strong>cluster</strong>
R package (available on CRAN) with the dataset name
<code>votes.repub</code>. The data from <code>1976-2020</code> are found
at <a href="https://doi.org/10.7910/DVN/42MVDX" class="external-link">Harvard Dataverse, US
President 1976-2020</a>. The second dataset has a CC0 1.0 Universal
license, and is included in the <strong>biwt</strong> R package as part
of the vignette.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># combining the two datasets and creating </span></span>
<span><span class="co"># a single data frame with the relevant information</span></span>
<span><span class="va">votes_repub</span> <span class="op">&lt;-</span> <span class="fu">cluster</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/cluster/man/votes.repub.html" class="external-link">votes.repub</a></span></span>
<span></span>
<span><span class="va">newer_president</span> <span class="op">&lt;-</span> <span class="fu">read_csv</span><span class="op">(</span><span class="st">"data/1976-2020-president.csv"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span>percent <span class="op">=</span> <span class="va">candidatevotes</span> <span class="op">/</span> <span class="va">totalvotes</span> <span class="op">*</span> <span class="fl">100</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">party_detailed</span> <span class="op">==</span> <span class="st">"REPUBLICAN"</span> <span class="op">&amp;</span> <span class="op">!</span><span class="va">writein</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">state</span> <span class="op">!=</span> <span class="st">"DISTRICT OF COLUMBIA"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">year</span>, <span class="va">state</span>, <span class="va">percent</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">pivot_wider</span><span class="op">(</span>id_cols <span class="op">=</span> <span class="va">state</span>, names_from <span class="op">=</span> <span class="va">year</span>, values_from <span class="op">=</span> <span class="va">percent</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Rows: </span><span style="color: #0000BB;">4287</span> <span style="font-weight: bold;">Columns: </span><span style="color: #0000BB;">15</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Column specification</span> <span style="color: #00BBBB;">────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="font-weight: bold;">Delimiter:</span> ","</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">chr</span> (6): state, state_po, office, candidate, party_detailed, party_simplified</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">dbl</span> (7): year, state_fips, state_cen, state_ic, candidatevotes, totalvotes, ...</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">lgl</span> (2): writein, notes</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use `spec()` to retrieve the full column specification for this data.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Specify the column types or set `show_col_types = FALSE` to quiet this message.</span></span>
<span></span>
<span><span class="va">all_votes_repub</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">votes_repub</span>, <span class="va">newer_president</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">state</span>, <span class="op">-</span><span class="va">X1976</span><span class="op">)</span> </span>
<span><span class="va">years</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"X"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">all_votes_repub</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># For states that did not exist or vote during certain years</span></span>
<span><span class="co"># (i.e. Alaska and Hawaii), the NA data is ignored.</span></span>
<span><span class="va">dend_NA</span> <span class="op">&lt;-</span> <span class="va">votes_repub</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">dist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html" class="external-link">as.dendrogram</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dendextend</span><span class="fu">::</span><span class="fu"><a href="http://talgalili.github.io/dendextend/reference/ladderize.html" class="external-link">ladderize</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">color_gradient</span> <span class="op">&lt;-</span> <span class="fu">colorspace</span><span class="fu">::</span><span class="va"><a href="https://colorspace.R-Forge.R-project.org/reference/hcl_palettes.html" class="external-link">diverge_hcl</a></span></span></code></pre></div>
<div class="section level3">
<h3 id="dendrograms">Dendrograms<a class="anchor" aria-label="anchor" href="#dendrograms"></a>
</h3>
<p>Dendrograms are a visualization of a hierarchical clustering
analysis. The hierarchical clustering agglomerates observations (here,
states) one at a time based on which states are closest to one another.
The dendrogram displays the entire agglomeration algorithm with shorter
vertical bars indicating states (or groups of states) that are closest
to one another and longer vertical bars indicating states (or groups of
states) that are farther from one another. In our work we use complete
linkage. The result is to identify which states had similar voting
patterns (measured by percent of Republican vote) across almost 200
years of presidential races.</p>
<p>The biweight correlation distance measure is the metric which seems
to cluster the states in a way that is most similar to how the politics
across states is considered. That is, the biweight correlation distance
was the best model option.</p>
<div class="section level4">
<h4 id="biweight-dendrogram">Biweight dendrogram<a class="anchor" aria-label="anchor" href="#biweight-dendrogram"></a>
</h4>
<p>The biweight correlation distance (between two states) is calculated
as one minus the biweight correlation between those two states. We can
observe that there are two quite distinct groups of states: the states
in the first arm (left side) are primarily states that are now
considered left leaning or “blue”; the states in the second arm (right
side) are primarily stats that are now considered right leaning or
“red.” The biweight distance measure does an excellent job of
distinguishing between the two groups of states.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dist.repub</span> <span class="op">&lt;-</span> <span class="va">all_votes_repub</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/biwt.cor.html">biwt.cor</a></span><span class="op">(</span>r <span class="op">=</span> <span class="fl">0.2</span>, absval <span class="op">=</span> <span class="cn">FALSE</span>, output <span class="op">=</span> <span class="st">"distance"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">as.dist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">usedist</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/usedist/man/dist_setNames.html" class="external-link">dist_setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">votes_repub</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"complete"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html" class="external-link">as.dendrogram</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span>ylab <span class="op">=</span> <span class="st">"biweigtht correlation distance"</span><span class="op">)</span></span></code></pre></div>
<p><img src="biwt_files/figure-html/unnamed-chunk-19-1.png" width="900" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="euclidean-dendrogram">Euclidean dendrogram<a class="anchor" aria-label="anchor" href="#euclidean-dendrogram"></a>
</h4>
<p>Euclidean distance is calculated using the Euclidean distance of the
Republican voting shares from 1856 to 2020. There are not clear
distinctions in the dendrogram to identify groups of states which are
most similar. In breaking the states up into clusters, 2 groups would
produce almost all states in the same cluster (except MS, SC, FL, AK,
TX, LA, AL, and GA). While the smaller of the two clusters might be
meaningfully related, it seems as though linking the other 42 states
might be too broad. However, how to break the dendrogram into additional
cluster (3 clusters? 4 clusters? 5 clusters?) is not an obvious
task.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dist.repub</span> <span class="op">&lt;-</span> <span class="va">all_votes_repub</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">dist</a></span><span class="op">(</span><span class="op">)</span><span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"complete"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html" class="external-link">as.dendrogram</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span>ylab <span class="op">=</span> <span class="st">"Euclidean distance"</span><span class="op">)</span></span></code></pre></div>
<p><img src="biwt_files/figure-html/unnamed-chunk-20-1.png" width="900" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="pearson-dendrogram">Pearson dendrogram<a class="anchor" aria-label="anchor" href="#pearson-dendrogram"></a>
</h4>
<p>The Pearson distance (between two states) is calculated as one minus
the Pearson correlation between those two states. Again, like the
Euclidean distance dendrogram, there are not obvious cutoff values to
determine the best number of clusters. Possibly, there could be 6 or 7
clusters, but the groupings do not necessarily align with current
understanding of political alignment across states.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pcor_dist</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">data</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fl">1</span> <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/cor.html" class="external-link">cor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">data</span><span class="op">)</span>, use <span class="op">=</span> <span class="st">"pairwise.complete.obs"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span></span>
<span><span class="va">dist.repub</span> <span class="op">&lt;-</span> <span class="va">all_votes_repub</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">pcor_dist</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">as.dist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">usedist</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/usedist/man/dist_setNames.html" class="external-link">dist_setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">votes_repub</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"complete"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html" class="external-link">as.dendrogram</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span>ylab <span class="op">=</span> <span class="st">"Pearson correlation distance"</span><span class="op">)</span></span></code></pre></div>
<p><img src="biwt_files/figure-html/unnamed-chunk-21-1.png" width="900" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level3">
<h3 id="heatmaps">Heatmaps<a class="anchor" aria-label="anchor" href="#heatmaps"></a>
</h3>
<p>To expand on the dendrogram, we add heatmaps to help visualize the
similarities and differences among states. Each color cell represents
the observed value given in the dataset (proportion of votes for the
Republican candidate). As shown in the key, the bluer a box is, the
fewer votes the republican candidate received. One note is that it is
important to note that this does not mean the rest of the votes were
given to the democrat candidate because there were still votes cast, for
example, for the Green Party candidate and write-in candidates.</p>
<p>The heatmap examples were heavily inspired by the <a href="Cluster%20Analysis%20vignette">https://cran.r-project.org/web/packages/dendextend/vignettes/Cluster_Analysis.html</a>
for the <a href="https://cran.r-project.org/web/packages/dendextend/vignettes/dendextend.html" class="external-link"><strong>dendextend</strong>
R package</a>.</p>
<div class="section level4">
<h4 id="biweight-heatmap">Biweight Heatmap<a class="anchor" aria-label="anchor" href="#biweight-heatmap"></a>
</h4>
<p>The biweight correlation is a good method for clustering this data
set because we can see the distinct separation between two groups of
states. Specifically looking from 2000 to 2020, the heatmap depicts
around half of the states mostly voting for the republican candidate and
the other half voting in other areas.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">bdend</span> <span class="op">&lt;-</span> <span class="va">all_votes_repub</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="../reference/biwt.cor.html">biwt.cor</a></span><span class="op">(</span>r <span class="op">=</span> <span class="fl">0.2</span>, absval <span class="op">=</span> <span class="cn">FALSE</span>, output <span class="op">=</span> <span class="st">"distance"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">as.dist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">usedist</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/usedist/man/dist_setNames.html" class="external-link">dist_setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">all_votes_repub</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"complete"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html" class="external-link">as.dendrogram</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dendextend</span><span class="fu">::</span><span class="fu"><a href="http://talgalili.github.io/dendextend/reference/rotate.html" class="external-link">rotate</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/labels.html" class="external-link">labels</a></span><span class="op">(</span><span class="va">dend_NA</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dendextend</span><span class="fu">::</span><span class="fu"><a href="http://talgalili.github.io/dendextend/reference/color_branches.html" class="external-link">color_branches</a></span><span class="op">(</span>k<span class="op">=</span><span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">gplots</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gplots/man/heatmap.2.html" class="external-link">heatmap.2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">all_votes_repub</span><span class="op">)</span>,</span>
<span>          main <span class="op">=</span> <span class="st">"Votes for\n Republican Presidential Candidate"</span>,</span>
<span>          srtCol <span class="op">=</span> <span class="fl">60</span>,</span>
<span>          dendrogram <span class="op">=</span> <span class="st">"row"</span>,</span>
<span>          Rowv <span class="op">=</span> <span class="va">bdend</span>,</span>
<span>          Colv <span class="op">=</span> <span class="st">"NA"</span>, <span class="co"># this to make sure the columns are not ordered</span></span>
<span>          trace<span class="op">=</span><span class="st">"none"</span>,          </span>
<span>          key.xlab <span class="op">=</span> <span class="st">"% Votes for Republican\n Presidential Candidate"</span>,</span>
<span>          labCol <span class="op">=</span> <span class="va">years</span>,</span>
<span>          denscol <span class="op">=</span> <span class="st">"grey"</span>,</span>
<span>          density.info <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>          keysize <span class="op">=</span> <span class="fl">1</span>,</span>
<span>          key.title <span class="op">=</span> <span class="st">""</span>,</span>
<span>          col <span class="op">=</span> <span class="va">color_gradient</span></span>
<span>         <span class="op">)</span></span></code></pre></div>
<p><img src="biwt_files/figure-html/unnamed-chunk-22-1.png" width="800" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="euclidean-heatmap">Euclidean Heatmap<a class="anchor" aria-label="anchor" href="#euclidean-heatmap"></a>
</h4>
<p>In the earlier years, each state’s voting pattern is well correlated
to states with similar voting patterns. Looking at the later years, the
voting patterns are more diverse in each correlation group. For example,
the blue group has a variety of states with high percentages of votes
for the republican candidate and other states with low percentages from
2000 to 2020.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">edend</span> <span class="op">&lt;-</span> <span class="va">votes_repub</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">dist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"complete"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html" class="external-link">as.dendrogram</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dendextend</span><span class="fu">::</span><span class="fu"><a href="http://talgalili.github.io/dendextend/reference/rotate.html" class="external-link">rotate</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/labels.html" class="external-link">labels</a></span><span class="op">(</span><span class="va">dend_NA</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dendextend</span><span class="fu">::</span><span class="fu"><a href="http://talgalili.github.io/dendextend/reference/color_branches.html" class="external-link">color_branches</a></span><span class="op">(</span>k<span class="op">=</span><span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">gplots</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gplots/man/heatmap.2.html" class="external-link">heatmap.2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">all_votes_repub</span><span class="op">)</span>,</span>
<span>          main <span class="op">=</span> <span class="st">"Votes for\n Republican Presidential Candidate"</span>,</span>
<span>          srtCol <span class="op">=</span> <span class="fl">60</span>,</span>
<span>          dendrogram <span class="op">=</span> <span class="st">"row"</span>,</span>
<span>          Rowv <span class="op">=</span> <span class="va">edend</span>,</span>
<span>          Colv <span class="op">=</span> <span class="st">"NA"</span>, <span class="co"># this to make sure the columns are not ordered</span></span>
<span>          trace<span class="op">=</span><span class="st">"none"</span>,          </span>
<span>          key.xlab <span class="op">=</span> <span class="st">"% Votes for Republican\n Presidential Candidate"</span>,</span>
<span>          labCol <span class="op">=</span> <span class="va">years</span>,</span>
<span>          denscol <span class="op">=</span> <span class="st">"grey"</span>,</span>
<span>          density.info <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>          keysize <span class="op">=</span> <span class="fl">1</span>,</span>
<span>          key.title <span class="op">=</span> <span class="st">""</span>,</span>
<span>          col <span class="op">=</span> <span class="va">color_gradient</span></span>
<span>         <span class="op">)</span></span></code></pre></div>
<p><img src="biwt_files/figure-html/unnamed-chunk-23-1.png" width="800" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="pearson-heatmap">Pearson Heatmap<a class="anchor" aria-label="anchor" href="#pearson-heatmap"></a>
</h4>
<p>When looking at the most recent years (2000 to 2020), the states are
well correlated, except the green group which has a good amount of mixed
percentages where some states voted more for the republicans and others
did not. The rest of the heatmap does not appear well correlated between
states and is spread out.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mar <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">pdend</span> <span class="op">&lt;-</span> <span class="va">all_votes_repub</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">pcor_dist</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">as.dist</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">usedist</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/usedist/man/dist_setNames.html" class="external-link">dist_setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">all_votes_repub</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/hclust.html" class="external-link">hclust</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"complete"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/dendrogram.html" class="external-link">as.dendrogram</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dendextend</span><span class="fu">::</span><span class="fu"><a href="http://talgalili.github.io/dendextend/reference/rotate.html" class="external-link">rotate</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/labels.html" class="external-link">labels</a></span><span class="op">(</span><span class="va">dend_NA</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dendextend</span><span class="fu">::</span><span class="fu"><a href="http://talgalili.github.io/dendextend/reference/color_branches.html" class="external-link">color_branches</a></span><span class="op">(</span>k<span class="op">=</span><span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">gplots</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/gplots/man/heatmap.2.html" class="external-link">heatmap.2</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">all_votes_repub</span><span class="op">)</span>,</span>
<span>          main <span class="op">=</span> <span class="st">"Votes for\n Republican Presidential Candidate"</span>,</span>
<span>          srtCol <span class="op">=</span> <span class="fl">60</span>,</span>
<span>          dendrogram <span class="op">=</span> <span class="st">"row"</span>,</span>
<span>          Rowv <span class="op">=</span> <span class="va">pdend</span>,</span>
<span>          Colv <span class="op">=</span> <span class="st">"NA"</span>, <span class="co"># this to make sure the columns are not ordered</span></span>
<span>          trace<span class="op">=</span><span class="st">"none"</span>,          </span>
<span>          key.xlab <span class="op">=</span> <span class="st">"% Votes for Republican\n Presidential Candidate"</span>,</span>
<span>          labCol <span class="op">=</span> <span class="va">years</span>,</span>
<span>          denscol <span class="op">=</span> <span class="st">"grey"</span>,</span>
<span>          density.info <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>          keysize <span class="op">=</span> <span class="fl">1</span>,</span>
<span>          key.title <span class="op">=</span> <span class="st">""</span>,</span>
<span>          col <span class="op">=</span> <span class="va">color_gradient</span></span>
<span>         <span class="op">)</span></span></code></pre></div>
<p><img src="biwt_files/figure-html/unnamed-chunk-24-1.png" width="800" style="display: block; margin: auto;"></p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Johanna Hardin.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
